288 update the code examples to use hypercore rather than compression

billy-the-fish opened the pull request.

billy-the-fish committed:
chore: first doc

billy-the-fish committed:
chore: sorting out new and deprecated features.

billy-the-fish committed:
chore: first doc.

billy-the-fish committed:
chore: migrate Use a job to downsample and compress chunks

billy-the-fish committed:
chore: migrate Use a job to downsample and compress chunks

billy-the-fish committed:
chore: update psql page

billy-the-fish committed:
chore: update blockchain tutorial

billy-the-fish committed:
chore: update blockchain tutorial

billy-the-fish committed:
chore: update Analyze financial tick data tutorial

billy-the-fish committed:
chore: update dual-write and backfill

billy-the-fish committed:
chore: last api check to replace compression apis with hypercore ones.

atovpeko commented on _partials/_create-hypertable-blockchain.md:
```diff
@@ -1,15 +1,20 @@
-standard PostgreSQL tables, indexes, stored procedures and other objects can be
-created alongside your Timescale hypertables. This makes creating and working
-with Timescale tables similar to standard PostgreSQL.
+$HYPERTABLE_CAP are the core of $TIMESCALE_DB. $HYPERTABLE_CAP enable $TIMESCALE_DB to work
```
```suggestion
$HYPERTABLE_CAPs are the core of $TIMESCALE_DB. $HYPERTABLE_CAPs enable $TIMESCALE_DB to work
```

atovpeko commented on _partials/_tutorials-hypercore-intro.md:
```diff
@@ -0,0 +1,21 @@
+Over time you end up with a lot of data. Since this data is mostly immutable you can compress it
```
```suggestion
Over time you end up with a lot of data. Since this data is mostly immutable, you can compress it
```

atovpeko commented on _partials/_tutorials-hypercore-intro.md:
```diff
@@ -0,0 +1,21 @@
+on the data and configuration.
+
+[$HYPERCORE_CAP][hypercore] is implemented natively in $PG and does not require special storage
+formats. When you convert your data from the $ROWSTORE to the $COLUMNSTORE. $TIMESCALE_DB uses
```
```suggestion
formats. When you convert your data from the $ROWSTORE to the $COLUMNSTORE, $TIMESCALE_DB uses
```

erimatnor committed:
Do not recommend setting Hypercore TAM on hypertable root (#4067)

Update the docs to recommend converting chunks to Hypercore TAM using
the columnstore policy instead of enabling it on new chunks.

Using Hypercore TAM on new chunks is suboptimal because there will be
no data compressed anyway until the policy runs, leading to a number
of confusing issues, such as no updated columnstore stats.

Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko commented on _partials/_tutorials-hypercore-intro.md:
```diff
@@ -0,0 +1,21 @@
+$PG features to transform the data into columnar format. The use of a columnar format allows a better
+compression ratio since similar data is stored adjacently. For more details on how
+the compression format looks, see [$HYPERCORE][hypercore].
+
```
I would rephrase to a more standard "For more details about the columnar format, see ...  "

atovpeko commented on _partials/_tutorials-hypercore-intro.md:
```diff
@@ -0,0 +1,21 @@
+compression ratio since similar data is stored adjacently. For more details on how
+the compression format looks, see [$HYPERCORE][hypercore].
+
+A beneficial side-effect of compressing data is that certain queries are significantly faster, since
```
```suggestion
A beneficial side effect of compressing data is that certain queries are significantly faster, since
```

atovpeko commented on migrate/dual-write-and-backfill/dual-write-from-postgres.md:
```diff
@@ -127,15 +127,15 @@ You may also wish to consider taking advantage of some of Timescale's killer
 - [retention policies] to automatically drop unneeded data
 - [tiered storage] to automatically move data to Timescale's low-cost bottomless object storage tier
-- [compression] to reduce the size of your hypertables
+- [hypercore] reduce the size of your hypertables by compressing data in the columnstore
```
```suggestion
- [hypercore] to reduce the size of your hypertables by compressing data in the columnstore
```

atovpeko commented on migrate/dual-write-and-backfill/timescaledb-backfill.md:
```diff
@@ -23,7 +23,7 @@ The `timescaledb-backfill` tool is a command-line utility designed to support
 to another ("backfilling"). `timescaledb-backfill` efficiently copies
 hypertable and continuous aggregates chunks directly, without the need for
-intermediate storage or decompressing compressed chunks. It operates
+intermediate storage, or moving chunks from $COLUMNSTORE to the $ROWSTORE. It operates
```
```suggestion
intermediate storage, or moving chunks from the $COLUMNSTORE to the $ROWSTORE. It operates
```

atovpeko commented on tutorials/blockchain-query/blockchain-compress.md:
```diff
@@ -97,71 +81,49 @@ table, you follow these step:
+
+## Take advantage of query speedups
+
+Previously, data in the $COLUMNSTORE was `segmentby` by the `block_id` column value.
```
```suggestion
Previously, data in the $COLUMNSTORE was segmented by the `block_id` column value.
```

atovpeko commented on tutorials/financial-tick-data/financial-tick-compress.md:
```diff
@@ -8,77 +8,70 @@ layout_components: [next_prev_large]
-1.  Connect to the Timescale database that contains the financial tick
-    dataset using, for example `psql`.
-1.  Enable compression on the table and pick suitable segment-by and
+1.  Enable the $COLUMNSTORE on the table and pick suitable segment-by and
```
Can we agree on one way to do this, please? Is it order-by/segment-by column or `orderby`/`segmentby` column?

atovpeko commented on tutorials/financial-tick-data/financial-tick-compress.md:
```diff
@@ -89,68 +82,44 @@ memory.
 
-You can find more information on compression policies in the
-[add_compression_policy][add_compression_policy] section.
+Previously, data in the $COLUMNSTORE was `segmentby` by the `block_id` column value.
```
```suggestion
Previously, data in the $COLUMNSTORE was segmented by by the `block_id` column value.
```

atovpeko commented on use-timescale/jobs/example-downsample-and-compress.md:
```diff
@@ -7,23 +7,23 @@ keywords: [jobs, compression, downsample]
-1.  Create a procedure that first queries the $CHUNKs of a $HYPERTABLE to
+1.  **Create a procedure to downsample chunks and convert them to columnstore**
+
+    This procedure that first queries the $CHUNKs of a $HYPERTABLE to
```
```suggestion
    This procedure first queries the $CHUNKs of a $HYPERTABLE to
```

atovpeko commented on use-timescale/continuous-aggregates/compression-on-continuous-aggregates.md:
```diff
@@ -5,83 +5,75 @@ products: [cloud, mst, self_hosted]
-columns of the continuous aggregate and the `time_bucket` column will be used as
-the `order_by` column in the compression configuration.
+$COLUMNSTORE_CAP works in the same way on [$HYPERTABLEs and $CAGGs][hypercore]. When you enable
+$COLUMNSTORE with no other options, your data is [segmentby][alter_materialized_view_arguments] the group by columns 
```
```suggestion
$COLUMNSTORE with no other options, your data is [segmented by][alter_materialized_view_arguments] the `groupby` columns 
```

atovpeko commented on use-timescale/continuous-aggregates/compression-on-continuous-aggregates.md:
```diff
@@ -5,83 +5,75 @@ products: [cloud, mst, self_hosted]
-the `order_by` column in the compression configuration.
+$COLUMNSTORE_CAP works in the same way on [$HYPERTABLEs and $CAGGs][hypercore]. When you enable
+$COLUMNSTORE with no other options, your data is [segmentby][alter_materialized_view_arguments] the group by columns 
+in the $CAGG, and [orderby][alter_materialized_view_arguments] the time column. [Real-time aggregation][real-time-aggregates]
```
```suggestion
in the $CAGG, and [ordered by][alter_materialized_view_arguments] the time column. [Real-time aggregation][real-time-aggregates]
```

atovpeko commented on use-timescale/continuous-aggregates/compression-on-continuous-aggregates.md:
```diff
@@ -5,83 +5,75 @@ products: [cloud, mst, self_hosted]
 
-1.  For an existing continuous aggregate, at the `psql` prompt, enable
-    compression:
+   To enable or disable compression on a $CAGG, set `timescaledb.enable_columnstore` when you alter the view.
```
```suggestion
   To enable the $COLUMNSTORE compression on a $CAGG, set `timescaledb.enable_columnstore = true` when you alter the view:
```

atovpeko commented on use-timescale/continuous-aggregates/compression-on-continuous-aggregates.md:
````diff
@@ -5,83 +5,75 @@ products: [cloud, mst, self_hosted]
+   ```sql
+   ALTER MATERIALIZED VIEW <cagg_name> set (timescaledb.enable_columnstore = true,);
+   ```
+   To Disable $COLUMNSTORE, set  `timescaledb.enable_columnstore = false`.
````
```suggestion
   To disable the $COLUMNSTORE compression, set  `timescaledb.enable_columnstore = false`:
```

atovpeko commented on use-timescale/continuous-aggregates/compression-on-continuous-aggregates.md:
```diff
@@ -5,83 +5,75 @@ products: [cloud, mst, self_hosted]
-decompress the chunks, and then drop any compression policy on the continuous
-aggregate, before you disable compression. For more detailed information, see
-the [decompress chunks][decompress-chunks] section:
+   Before you setup a $COLUMNSTORE policy on a $CAGG, you first set the [refresh policy][refresh-policy]. To 
```
```suggestion
   Before you set up a $COLUMNSTORE policy on a $CAGG, you first set the [refresh policy][refresh-policy]. To 
```

atovpeko commented:
I think we are also missing a redirect here?

billy-the-fish committed:
chore: updates on the windows install.  (#4063)

* chore: updates

* Update _troubleshooting/self-hosted/windows-install-library-not-loaded.md

Co-authored-by: Jônatas Davi Paganini <jonatas@timescale.com>
Signed-off-by: Iain Cox <iain@timescale.com>

---------

Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Jônatas Davi Paganini <jonatas@timescale.com>

coelhucas committed:
chore: replace repository dispatch with workflow dispatch (#4079)

* chore: replace repository dispatch with workflow dispatch

* chore: add missing parameter workflow

* chore: update ref parameter

* chore: trigger workflow

* chore: remove commented out code

coelhucas committed:
chore: change ref branch to production-v3 (#4087)

coelhucas committed:
chore: replace deprecated action by maintained fork (#4091)

* chore: replace deprecated action by maintained fork

* chore: add config to not report too many requests

billy-the-fish committed:
chore: remove enable_segmentwise_recompression (#4092)

billy-the-fish committed:
Apply suggestions from code review

Co-authored-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>
Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
chore: updates on review.

billy-the-fish commented:
@atovpeko  for which file? https://docs.timescale.com/use-timescale/latest/continuous-aggregates/compression-on-continuous-aggregates/ already exists.

billy-the-fish committed:
Merge branch '289-milestone-branch-for-220' into 288-update-the-code-examples-to-use-hypercore-rather-than-compression

Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish commented on _partials/_create-hypertable-blockchain.md:
```diff
@@ -1,15 +1,20 @@
 <Procedure>
 
-1.  Create a standard PostgreSQL table to store the Bitcoin blockchain data
+1. Connect to your $SERVICE_LONG
```
For reviewers, this example is updated to use CREATE TABLE In another PR: https://github.com/timescale/docs/pull/4089/files#diff-5953950de370451ab0d4f26a8324e4164fdf2bba054442ce754f957332b93b34. Please just ignore the code here.

billy-the-fish committed:
Merge branch '289-milestone-branch-for-220' into 288-update-the-code-examples-to-use-hypercore-rather-than-compression

Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish commented:
To unify the changes, i've moved the doc updates to https://github.com/timescale/docs/pull/4089/files.
